extensions/wikia/InsightsV2/services/InsightsService.class.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+
+class InsightsService extends WikiaService
+{
+
+	/**
+	 * @param string $type type of model
+	 * @param int $size number of pages we need
+	 * @return array
+	 */
+	public function getInsightPages( $type, $size, $sortingType ) {
+		if ( InsightsHelper::isInsightPage( $type ) ) {
+			$model = InsightsHelper::getInsightModel( $type );
+			$insightData = ( new InsightsContext( $model ) )->fetchData();
+			$sortedInsightArticleIds = ( new InsightsSorting( $model->getConfig() ) )->getSortedData( $insightData, ['sort' => $sortingType] );
+			$aritclesIds = $this->truncateTo( $sortedInsightArticleIds, $size );
+			return $this->getArticlesData( $insightData, $aritclesIds );
+		}
+		return [];
```

Comment:
I'd consider reversing this so the early return of empty data is in an if block, and the main work of the method is unindented.